monai/transforms/utils.py

```diff
@@ -211,6 +211,51 @@ def map_binary_to_indices(
     return fg_indices, bg_indices
 
 
+def weighted_patch_samples(
```

Comment:
I used something similar to this before:

```py
def select_coords(w,n_samples,r_state):
    w = np.clip(w, 0, 1)
    coords = list(np.ndindex(*w.shape))
    
    selections= r_state.choice(len(coords), n_samples, p=w.flatten()/w.sum())
    
    return [coords[idx] for idx in selections]
```

It's not the equivalent as here but perhaps `choice` could simplify the implementation.

Reply:
that API requires normalised `p`, I don't think it's necessary for our case